src/renderers/WebGLRenderer.js

```diff
@@ -1036,15 +1004,15 @@ class WebGLRenderer {
 
 							const material2 = material[ i ];
 
-							prepareMaterial( material2, scene, object );
-							compiling.add( material2 );
+							prepareMaterial( material2, targetScene, object );
```

Comment:
This is a subtle but important change. We have to always use `targetScene` here since otherwise we miss environment definitions (and then end up with recompilations during the first render).